liferay-frontend/liferay-portal
# LPS-166479 - Create new client extension for theme spritemap

This PR adds a new field to themeDisplay for accessing the spritemap and also adds a client extension to override this value. Icons are tied pretty closely to the theme, as is JS and CSS and so it didn't make much sense to continue with the "icon pack" idea as our client extensions are overriding these sorts of theme elements. I wasn't able to come up with a great solution that would both allow for icon packs and client extension for icon pack, they seemed to address the same concern in different ways. By moving the spritemap to themeDisplay we now have a singular entry point to access icons and also aligns the resource similarly to our css and js theme resources.

## How to test:
- Add FF "LPS-166479"
- Upload new SVG to document library
- Copy URL to file
- Create "SVG" client extension using the URL from the doc-lib
- Navigate to edit the "Look and Feel" of you page
- Select the registered "Theme Spritemap Client Extension"
- Save
- Go back to page and see new SVG used as spritemap
	- You can even inspect the page and see the the icons reference the spritemap in the document library


Here is an SVG file to use, I simply took the existing spritemap and swapped "cog" for "pencil". [icons.svg](https://user-images.githubusercontent.com/6843530/204792992-bed2c337-d017-40f5-ad13-5917ee5b129e.svg)

### Video Demo
https://user-images.githubusercontent.com/6843530/204792878-8ebc8921-6ec8-4aa3-b283-a0060a85920b.mp4
